xml.c

```diff
@@ -421,17 +423,42 @@ static struct iio_context * iio_create_xml_context_helper(xmlDoc *doc)
 	}
 
 	for (attr = root->properties; attr; attr = attr->next) {
-		if (!strcmp((char *) attr->name, "description"))
-			description = (const char *)attr->children->content;
-		else if (strcmp((char *) attr->name, "name"))
+		content = (const char *) attr->children->content;
+
+		if (!strcmp((char *) attr->name, "description")) {
+			description = content;
+		} else if (!strcmp((char *) attr->name, "version-major")) {
+			major = strtol(content, &end, 10);
+			if (*end != '\0')
+				IIO_WARNING("unable to parse major version\n");
```

Comment:
nit: I think this message might be a bit misleading. IIRC, this condition pretty much says that we have further characters on our string but the return value could actually be correct. eg; If the major string is something like "2foo", I think val would be 2 (which could be a valid value).

Reply:
`invalid format for major version` sounds better?